geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/security/Authenticator.java

```diff
@@ -33,27 +34,10 @@
    *
    * @param inputStream to read auth messages from.
    * @param outputStream to send messages to.
-   * @param securityManager can be used for validating credentials against.
+   * @param securityService used for validating credentials.
+   * @return authenticated principal
    * @throws IOException if EOF or if invalid input is received.
    */
-  void authenticate(InputStream inputStream, OutputStream outputStream,
-      SecurityManager securityManager) throws IOException;
-
-  /**
-   * Until authentication is complete, isAuthenticated() must return false, and the socket will
-   * always be passed to the Authenticator. Once authentication succeeds, calls to this function
-   * must always return true.
-   */
-  boolean isAuthenticated();
-
-  /**
-   * Return an authorization object which can be used to determine which permissions this stream has
-   * according to the provided securityManager.
-   */
-  Authorizer getAuthorizer() throws AuthenticationRequiredException;
-
-  /**
-   * @return a unique identifier for this particular implementation (NOOP, PASSTHROUGH, etc.)
-   */
-  String implementationID();
```

Comment:
why would we have removed the `implementationID`? How do we handle multiple authentication mechanisms?

Reply:
We now choose the authentication required based on the security settings.  The implementationID was just a way to map the authenticator to the property that we had added.

Reply:
Although perhaps renaming to `getImplementationID()` would be better...